### PR TITLE
fix(core): Frame to Page property propagation

### PR DESCRIPTION
**Pull request by CatchABus**
## PR Checklist

- [x] The PR title follows our guidelines: https://github.com/NativeScript/NativeScript/blob/master/tools/notes/CONTRIBUTING.md#commit-messages.
- [ ] There is an issue for the bug/feature this PR is for. To avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it.
- [x] You have signed the [CLA](http://www.nativescript.org/cla).
- [x] All existing tests are passing: https://github.com/NativeScript/NativeScript/blob/master/tools/notes/DevelopmentWorkflow.md#running-unit-tests-application.
- [ ] Tests for the changes are included - https://github.com/NativeScript/NativeScript/blob/master/tools/notes/WritingUnitTests.md.

## What is the current behavior?
When I attempt to update some inheritable properties programmatically, they won't propagate from Frame to Page during initialization.
The reason seems to be the fact that we append page to frame using `_addView` before setting new frame entry.
Methods like `eachChildView` and `_childrenCount` rely on current entry resolved page and this is the actual problem.

For example, this is a description of how things currently work on iOS:
- Method `viewWillAppear` in Frame view controller will perform an `_addView` call
- Method `viewDidAppear` will assign new current entry

Property `currentPage` is a getter that returns the resolved page of current entry, and this getter is used by `eachChildView` and `_childrenCount`. Since current entry won't be updated until `viewDidAppear` call, we end up with an `_addView` call that tries to iterate Frame's children in order to update their inheritable properties but `eachChildView` has still got zero iterations as `currentPage` returns null.

## What is the new behavior?
We add a new property called `_childView` in order to separate entry navigation logic from view tree logic. Methods like `eachChildView` are considered a DOM thing and we should not make them rely on navigation entries.

Tests seem good in both platforms but we might want to spend some time and see if this change has any bad impact.

### CODE REVIEW

**Comment by farfromrefug** [comment]
@CatchABus could we add a comment or rename the variable to express it is exactly the same as currentEntry.resolvedPage but only set sooner?

**Comment by CatchABus** [comment]
> @CatchABus could we add a comment or rename the variable to express it is exactly the same as currentEntry.resolvedPage but only set sooner?

Yes sure, would something like `_resolvedPage` be more descriptive? I originally went for `_currentPage` but its naming is very close to existing `currentPage` getter

**Comment by farfromrefug** [comment]
@CatchABus i like resolvedPage

File: packages/core/ui/frame/frame-common.ts [hunk 1]
```diff
@@ -43,6 +43,14 @@ export class FrameBase extends CustomLayoutView {
 
 	public actionBarVisibility: 'auto' | 'never' | 'always';
 	public _currentEntry: BackstackEntry;
+
+	/**
```
Comment (CatchABus) [inline packages/core/ui/frame/frame-common.ts:47]: @farfromrefug What do you think about this? Do we need more details about this change?

Comment (farfromrefug) [reply packages/core/ui/frame/frame-common.ts:47]: Seems good to.me!

**Review by farfromrefug — APPROVED** [review]

**Pull request merged**